tests/istio.io/tasks/security/mtls-migration/mtls-migration_test.go

```diff
@@ -34,42 +34,87 @@ func TestMain(m *testing.M) {
+	parts := strings.Split(lines[1], " ")
+	if len(parts) != 2 {
+		return fmt.Errorf("expected output to follow: namespace, name, age; actual was: %s", lines[1])
+	}
 
-	//verify that all requests returns 200 ok
+	if parts[0] != "istio-system" {
+		return fmt.Errorf("expected namespace to be istio-system actual: %s", parts[0])
+	}
 
-	ex.RunScript("verify-initial-policies.sh", examples.TextOutput)
+	if parts[1] != "grafana-ports-mtls-disabled" {
+		return fmt.Errorf("expected policies to be grafana-ports-mtls-disabled actual: %s", parts[1])
+	}
 
-	//verify that only the following exist:
-	// NAMESPACE      NAME                          AGE
-	// istio-system   grafana-ports-mtls-disabled   3m
-
-	ex.RunScript("verify-initial-destinationrules.sh", examples.TextOutput)
+	return nil
+}
 
+func validateInitialDestinationRules(output string) error {
 	//verify that only the following exists:
 	//NAMESPACE      NAME              AGE
 	//istio-system   istio-policy      25m
 	//istio-system   istio-telemetry   25m
 
-	ex.RunScript("configure-mtls-destinationrule.sh", examples.TextOutput)
-	ex.RunScript("curl-foo-bar-legacy.sh", examples.TextOutput)
+	lines := strings.Split(output, "\n")
+	if len(lines) < 3 {
+		return fmt.Errorf("expected output to be 3 lines; actual: %d", len(lines))
+	}
 
-	//verify 200ok from all requests
+	line := lines[1]
+	parts := strings.Split(line, " ")
+	if len(parts) != 2 {
+		return fmt.Errorf("expected output to follow namespace, name, age; actual was: %s", line)
+	}
 
-	ex.RunScript("httpbin-foo-mtls-only.sh", examples.TextOutput)
-	ex.RunScript("curl-foo-bar-legacy.sh", examples.TextOutput)
+	if parts[0] != "istio-system" {
+		return fmt.Errorf("expected namespace to be istio-system; actual: %s", parts[0])
+	}
 
-	//verify 200 from first 2 requests and 503 from 3rd request
+	if parts[1] != "istio-policy" {
+		return fmt.Errorf("expected name to be istio-policy; actual: %s", parts[1])
+	}
 
-	ex.RunScript("cleanup.sh", examples.TextOutput)
-	ex.Run()
+	line = lines[2]
+	parts = strings.Split(line, " ")
+	if parts[0] != "istio-system" {
+		return fmt.Errorf("expected namespace to be istio-system; actual: %s", parts[0])
+	}
+
+	if parts[1] != "istio-policy" {
+		return fmt.Errorf("expected name to be istio-telemetry; actual: %s", parts[2])
+	}
+
+	return nil
+}
+
+//https://istio.io/docs/tasks/security/mtls-migration/
+//https://github.com/istio/istio.io/blob/release-1.2/content/docs/tasks/security/mtls-migration/index.md
+func TestMTLS(t *testing.T) {
+	//Test
+	examples.New(t, "mtls-migration").
+		RunScript("create-ns-foo-bar-legacy.sh", examples.TextOutput, nil).
+		WaitForPods(examples.NewMultiPodFetch("foo")).
+		WaitForPods(examples.NewMultiPodFetch("bar")).
+		WaitForPods(examples.NewMultiPodFetch("legacy")).
+		RunScript("curl-foo-bar-legacy.sh", examples.TextOutput, examples.GetCurlVerifier([]string{"200", "200", "200"})).
+		//RunScript("verify-initial-policies.sh", examples.TextOutput, validateInitialPolicies).
```

Comment:
These two lines are currently commented because the environment created by framework.NewSuite does not match the expected prerequisites. I see these in minikube though and this test does pass with these commented out.

Reply:
@nmittler Nathan, any idea about what's breaking here?

Reply:
Looks like grafana is disabled in the [default values file](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/values.yaml#L65)

I suspect for examples, we should use the [demo values file](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/values-istio-demo.yaml#L58).

[Here](https://github.com/istio/istio/blob/master/tests/integration/security/sds_citadel_flow/main_test.go#L65) is an example of how to specify the values file using the test framework.